internal/publisher/core_test.go

```diff
@@ -0,0 +1,74 @@
+package publisher
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	brokerstoremock "github.com/razorpay/metro/internal/brokerstore/mocks"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	mockMB "github.com/razorpay/metro/pkg/messagebroker/mocks"
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+	"github.com/rs/xid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCore_Publish(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	mockBrokerStore := brokerstoremock.NewMockIBrokerStore(ctrl)
+	producer := mockMB.NewMockProducer(ctrl)
+	publisherCore := NewCore(mockBrokerStore)
+	req := getDummyPublishRequest()
+	msgID := xid.New().String()
+
+	tests := []struct {
```

Comment:
you include name of test in test struct  and run test case like `for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {}}`
this will log test name if test case is failing

Reply:
test names are logged even without the above statement. You can check it on github as well when the unit tests are failing as well as on local

Reply:
the test with multiple test we will not be able to get which of those cases are missing

Reply:
<img width="1459" alt="Screenshot 2022-09-14 at 4 18 42 PM" src="https://user-images.githubusercontent.com/104060154/191663032-d38f659a-379e-46e7-8aa5-f874bdd8bdc1.png">

Multiple tests case are failing. Do you have some doc that says otherwise..?

Reply:
If test have multiple test cases we will be able to identify which case is failing